torch_xla/csrc/convolution.cpp

```diff
@@ -98,6 +98,34 @@ namespace {
  */
 // clang-format on
 
+// Input needs to be NCHW format.
+xla::XlaOp padInputFromOutputSize(xla::XlaOp input,
+                                  absl::Span<const xla::int64> stride,
+                                  absl::Span<const xla::int64> output_padding,
+                                  bool unpad = false) {
+  const xla::Shape& input_shape = XlaHelpers::ShapeOfXlaOp(input);
+  xla::int64 num_spatial = input_shape.rank() - 2;
+  xla::PaddingConfig padding_config;
+  // Only pad the spatial dimensions.
+  padding_config.add_dimensions();
+  padding_config.add_dimensions();
+  for (int spatial_dim = 0; spatial_dim < num_spatial; ++spatial_dim) {
+    xla::int64 input_spatial = input_shape.dimensions(2 + spatial_dim);
+    // Input_spatial needs to increase by pad_to_input to generate the output
+    // that includes output_padding. The formula is derived from the output size
+    // calculation in the BuildTransposedConvolution.
+    xla::int64 pad_to_input = ((input_spatial - 1) * stride[spatial_dim] +
+                               output_padding[spatial_dim]) /
+                                  stride[spatial_dim] +
+                              1 - input_spatial;
+    xla::PaddingConfig::PaddingConfigDimension* dims =
+        padding_config.add_dimensions();
+    dims->set_edge_padding_high(unpad ? -1 * pad_to_input : pad_to_input);
+  }
+  xla::XlaOp zero = xla::Zero(input.builder(), input_shape.element_type());
+  return xla::Pad(input, zero, padding_config);
```

Comment:
I'd use the `PadToSize()` API here.
Just added a small optimization.

https://github.com/pytorch/xla/pull/2352